SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp

```diff
@@ -508,3 +535,43 @@ void USpatialClassInfoManager::QuitGame()
 	FGenericPlatformMisc::RequestExit(false);
 #endif
 }
+
+Worker_ComponentId USpatialClassInfoManager::ComputeActorInterestComponentId(const AActor* Actor) const
+{
+	check(Actor);
+	const AActor* ActorForRelevancy = Actor;
+	// bAlwaysRelevant takes precedence over bNetUseOwnerRelevancy - see AActor::IsNetRelevantFor
+	while (!ActorForRelevancy->bAlwaysRelevant && ActorForRelevancy->bNetUseOwnerRelevancy && ActorForRelevancy->GetOwner())
+	{
+		ActorForRelevancy = ActorForRelevancy->GetOwner();
+	}
+
+	if (ActorForRelevancy->bAlwaysRelevant)
+	{
+		return SpatialConstants::ALWAYS_RELEVANT_COMPONENT_ID;
+	}
+
+	if (GetDefault<USpatialGDKSettings>()->bEnableNetCullDistanceInterest)
+	{
+		Worker_ComponentId NCDComponentId = GetComponentIdForNetCullDistance(ActorForRelevancy->NetCullDistanceSquared);
+		if (NCDComponentId != SpatialConstants::INVALID_COMPONENT_ID)
+		{
+			return NCDComponentId;
+		}
+
+		const AActor* DefaultActor = Actor->GetClass()->GetDefaultObject<AActor>();
+		if (Actor != DefaultActor && Actor->NetCullDistanceSquared != DefaultActor->NetCullDistanceSquared)
+		{
+			UE_LOG(LogSpatialClassInfoManager, Error, TEXT("Could not find Net Cull Distance Component for distance %f, processing Actor %s, because its Net Cull Distance is different from its default one."),
+				Actor->NetCullDistanceSquared, *Actor->GetPathName());
+
+			return ComputeActorInterestComponentId(DefaultActor);
+		}
+		else
+		{
+			UE_LOG(LogSpatialClassInfoManager, Error, TEXT("Could not find Net Cull Distance Component for distance %f, processing Actor %s. Have you generated schema?"),
```

Comment:
Why is this an error? Does this not just indicate that the default NCD should be used? (And its corresponding component)

Reply:
It indicates that you game's schema hasn't been prepared correctly (ie. your actor class's NCD component wasn't generated). This would be user error, and means your game is badly prepared.